### PR TITLE
Refactor semesters to plans

### PR DESCRIPTION
**Pull request by elizabeth-tang**
**Summary**
Remove all uses of semesters in Vuex store and in Firebase data
- Replace references to semesters field in VuexStoreState with new schema using the new plan data type.
- Address edge case with rendering before Vuex store is set.

**Remaining TODOs:**

- Stub in rest of multiple plans functionality
**Test Plan**
- After running migration script on all users/just your username if testing, does the app load with 1 plan?
- Does adding a new plan show up for you on Firebase?
- Does adding a new semester or classes to a semester update the correct plan on Firebase?

### CODE REVIEW

File: src/components/Semester/SemesterView.vue [hunk 1]
```diff
@@ -107,7 +107,14 @@ export default defineComponent({
   },
   computed: {
     semesters(): readonly FirestoreSemester[] {
-      return store.state.semesters;
+      console.log(store.state.plans);
```
Comment (andxu282) [inline src/components/Semester/SemesterView.vue:110]: get rid of console log

File: src/user-data.d.ts [hunk 2]
```diff
@@ -77,15 +77,15 @@ type FirestoreCourseOptInOptOutChoices = {
...
+  readonly arbitraryOptIn: { readonly [requirement: string]: readonly string[] };
 };
 type FirestoreOverriddenFulfillmentChoices = {
   readonly [courseUniqueId: string]: FirestoreCourseOptInOptOutChoices;
 };
 
 type FirestoreUserData = {
   readonly name: FirestoreUserName;
-  readonly semesters: FirestoreSemester[];
+  // readonly semesters: readonlyFirestoreSemester[];
```
Comment (andxu282) [inline src/user-data.d.ts:88]: delete

File: src/store.ts [hunk 3]
```diff
@@ -283,15 +314,21 @@ export const initializeFirestoreListeners = (onLoad: () => void): (() => void) =
...
+      store.commit('setPlans', data.plans);
+      store.commit('setCurrentPlan', data.plans[0]);
+
+      const plan = getFirstPlan(data);
       const { orderByNewest } = data;
-      store.commit('setSemesters', semesters);
+      store.commit('setSemesters', plan.semesters);
       updateDoc(doc(fb.semestersCollection, simplifiedUser.email), {
-        plans: [{ semesters }], // TODO: andxu282 update later
+        plans: [plan], // TODO: andxu282 update later
```
Comment (andxu282) [inline src/store.ts:324]: can u get rid of comment 🥲

File: src/store.ts [hunk 4]
```diff
@@ -300,7 +337,7 @@ export const initializeFirestoreListeners = (onLoad: () => void): (() => void) =
       store.commit('setSemesters', [newSemester]);
       setDoc(doc(fb.semestersCollection, simplifiedUser.email), {
         orderByNewest: true,
-        plans: [{ semesters: [newSemester] }], // TODO: andxu282 update later
+        plans: [{ name: 'Plan 1', semesters: [newSemester] }], // TODO: andxu282 update later
```
Comment (andxu282) [inline src/store.ts:340]: remove comment : )

File: src/store.ts [hunk 5]
```diff
@@ -76,7 +78,7 @@ const store: TypedVuexStore = new TypedVuexStore({
       tookSwim: 'no',
     },
     orderByNewest: true,
-    semesters: [],
+    // semesters: [],
```
Comment (andxu282) [inline src/store.ts:81]: delete

File: src/store.ts [hunk 6]
```diff
@@ -36,7 +36,7 @@ export type VuexStoreState = {
   currentFirebaseUser: SimplifiedFirebaseUser;
   userName: FirestoreUserName;
   onboardingData: AppOnboardingData;
-  semesters: readonly FirestoreSemester[];
+  // semesters: readonly FirestoreSemester[];
```
Comment (andxu282) [inline src/store.ts:39]: delete

**Review by andxu282 — COMMENTED** [review]
Looking good, just some clean up items, I would do a command f for semesters just in case you missed something, did you ever end up fixing that bug btw?

**Review by zachary-kent — CHANGES_REQUESTED** [review]
I think it would probably be easier and cleaner if you modified the `semesters` getter to return `plans[currentPlan].semesters`. That way, accesses to `store.state.semesters` wouldn't have to be changed throughout the codebase.

**Review by zachary-kent — CHANGES_REQUESTED** [review]
Instead of passing the current plan into functions like `addCourseToSemester` and `deleteCourseFromSemesters`, can we import the vuex store and access the current plan directly inside of these functions?

**Review by zachary-kent — CHANGES_REQUESTED** [review]
Looks good! Can you delete the spurious console statements?

File: src/components/Semester/Semester.vue [hunk 7]
```diff
@@ -455,15 +470,28 @@ export default defineComponent({
       this.openConfirmationModal(`Added ${course.code} to ${this.season} ${this.year}`);
     },
     deleteCourseWithoutModal(uniqueID: number) {
-      deleteCourseFromSemester(this.year, this.season, uniqueID, this.$gtag);
+      deleteCourseFromSemester(
+        store.state.currentPlan,
```
Comment (zachary-kent) [inline src/components/Semester/Semester.vue:474]: Why do we need to pass the current plan as an argument?

**Review by zachary-kent — CHANGES_REQUESTED** [review]

File: src/components/Semester/Semester.vue (hunk 7 above)
Comment (elizabeth-gs) [reply src/components/Semester/Semester.vue:474]: I guess there's no case where you aren't deleting from the current plan currently but it was just the way I wrote the firestore functions to modify the plan that is passed in.

**Review by zachary-kent — CHANGES_REQUESTED** [review]
Multiple plans dropdown openclose (failed)
It looks like the test is failing because the dropdown isn't opened after it's clicked.
